plugins/com.google.cloud.tools.eclipse.appengine.libraries.test/xsd/com.google.cloud.tools.eclipse.appengine.libraries.xsd

```diff
@@ -0,0 +1,267 @@
+      <xs:appinfo>
+         <meta.section type="xs:copyright"/>
+      </xs:appinfo>
+      <xs:documentation>
+         Copyright 2016 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+      </xs:documentation>
+   </xs:annotation>
+
+   <xs:element name="plugin">
+      <xs:complexType>
+         <xs:choice minOccurs="1" maxOccurs="unbounded">
+            <xs:element ref="extension-point" />
+            <xs:element ref="extension" />
+         </xs:choice>
+      </xs:complexType>
+   </xs:element>
+
+   <xs:element name="extension-point">
+      <xs:complexType>
+         <xs:attribute name="id" type="xs:string" />
+         <xs:attribute name="name" type="xs:string" />
+         <xs:attribute name="schema" type="xs:string" />
+      </xs:complexType>
+   </xs:element>
+   
+   <xs:element name="extension">
+      <xs:annotation>
+         <xs:appinfo>
+            <xs:meta.element />
+         </xs:appinfo>
+      </xs:annotation>
+      <xs:complexType>
+         <xs:choice minOccurs="1" maxOccurs="unbounded">
+            <xs:element ref="library" />
+            <xs:element ref="classpathContainerInitializer" />
+         </xs:choice>
+         <xs:attribute name="point" type="xs:string" use="required" />
+         <xs:attribute name="id" type="xs:string" />
+         <xs:attribute name="name" type="xs:string">
+            <xs:annotation>
+               <xs:appinfo>
+                  <xs:meta.attribute translatable="true" />
+               </xs:appinfo>
+            </xs:annotation>
+         </xs:attribute>
+      </xs:complexType>
+   </xs:element>
+
+   <xs:element name="classpathContainerInitializer">
+      <xs:complexType>
+         <xs:attribute name="class" type="xs:string" />
+         <xs:attribute name="id" type="xs:string" />
+      </xs:complexType>
+   </xs:element>
+
+   <xs:element name="library">
+      <xs:annotation>
+         <xs:documentation>
+            An App Engine library that can consist of one or more jar files with associated visibility rules and Maven coordinates.
+         </xs:documentation>
+      </xs:annotation>
+      <xs:complexType>
+         <xs:choice minOccurs="1" maxOccurs="unbounded">
+            <xs:element ref="libraryFile" />
+         </xs:choice>
+         <xs:attribute name="id" type="xs:string">
+            <xs:annotation>
+               <xs:documentation>
+                  The ID of the library. The path of the classpath container defined for this library will be 
```

Comment:
You're putting implementation detail here that's unnecessary.  You could put it into the Implementation section of the extension point.